sonarqube/datadog_checks/sonarqube/discovery_matcher.py

```diff
@@ -0,0 +1,113 @@
+import re
+from typing import List
+
+from datadog_checks.base import ConfigurationError
+
+
+class DiscoveryMatcher:
```

Comment:
Can you add some documentation about what this class does?

Reply:
Done